test/unit/math/gpu/cholesky_decompose_test.cpp

```diff
@@ -0,0 +1,79 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/gpu/copy.hpp>
+#include <stan/math/gpu/cholesky_decompose.hpp>
+#include <stan/math/prim/mat/fun/cholesky_decompose.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/err/check_pos_definite.hpp>
+#include <stan/math/prim/mat/err/check_square.hpp>
+#include <stan/math/prim/mat/err/check_symmetric.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+#define EXPECT_MATRIX_NEAR(A, B, DELTA) \
+  for (int i = 0; i < A.size(); i++)    \
+    EXPECT_NEAR(A(i), B(i), DELTA);
+
+TEST(MathMatrix, cholesky_decompose_cpu_vs_gpu_small) {
+  stan::math::matrix_d m0(3, 3);
+  m0 << 25, 15, -5, 15, 18, 0, -5, 0, 11;
+
+  stan::math::matrix_d m1(4, 4);
+  m1 << 18, 22, 54, 42, 22, 70, 86, 62, 54, 86, 174, 134, 42, 62, 134, 106;
+
+  stan::math::matrix_gpu m0_gpu(m0);
+  stan::math::matrix_gpu m1_gpu(m1);
+
+  stan::math::matrix_d m0_res = stan::math::cholesky_decompose(m0);
+  stan::math::matrix_d m1_res = stan::math::cholesky_decompose(m1);
+
+  stan::math::matrix_gpu m0_chol_gpu = stan::math::cholesky_decompose(m0_gpu);
```

Comment:
With STAN_OPENCL defined this actually both use GPUs. In the first case directlyI , and in the second one through the cholesky_decompose(Eigen Matrix&) function. That is why I resorted to directly calling Eigens LLT in the _big tests. LLT is what is called in the cholesky_decompose CPU version. 

As we are only chanhing what is called underneath we could resort to only using the /prim/mat/fun/ tests. Although the cholesky tests only check for exceptions, not correct values.

Reply:
> might want to start thinking about how to test the GPU stuff mixed in with the rest of the math stuff (eg some light tests at the API level when including the standard include that the language uses

I agree, we can bring those tests in with the cholesky rev